### ISSUE TITLE
Foxx.Model static event listeners are not invoked by Foxx.Repository

### ISSUE THREAD

**Issue by pluma** [arangodb/arangodb#1665]
The static event listeners that can be defined on Foxx.Model sub-types are not invoked when the apropriate Foxx.Repository methods are called. The cause seems to be that `_.each` does not iterate over the properties of functions.

This affects 2.8 and 2.7 and likely also 2.6.

**Comment by CoDEmanX** [comment]
```js
var f = function(){};
f.foo = "bar";
console.log(f.length) // 0
console.log(Object.keys(f)) // ["foo"]
```

The function does have a length property, but it's zero. According to https://lodash.com/docs#forEach (which is the same as `_.each()`), the presence of the `length` property will make it iterate the object like an array. The length is 0, so nothing to iterate over. `_.forEach()` should probably be replaced by `_.forIn()` or rather `_.forOwn()`.

**Comment by pluma** [comment]
2.x uses underscore instead of lodash, so the best alternative seems to be `Object.keys(x).forEach`.

**Comment by CoDEmanX** [comment]
`_.mapObject()` perhaps?

**Comment by pluma** [comment]
The fix is in 2.6, 2.7, 2.8 and devel.
